File: sdk/typescript/src/namespaces/matrix-debates.ts
/**
 * Matrix Debates Namespace API
 *
 * Provides endpoints for matrix debate management.
 * Matrix debates enable multi-dimensional analysis across
 * multiple perspectives simultaneously.
 */

interface MatrixDebatesClientInterface {
  request<T = unknown>(
    method: string,
    path: string,
    options?: { params?: Record<string, unknown>; json?: Record<string, unknown> }
  ): Promise<T>;
}

/**
 * Matrix Debates API namespace.
 *
 * @example
 * ```typescript
 * const debates = await client.matrixDebates.list();
 * const debate = await client.matrixDebates.get(debates[0].id);
 * ```
 */
export class MatrixDebatesAPI {
  constructor(private client: MatrixDebatesClientInterface) {}

  /**
   * List matrix debates.
   * @route GET /api/v1/matrix-debates
   */
  async list(params?: Record<string, unknown>): Promise<Record<string, unknown>> {
    return this.client.request('GET', '/api/v1/matrix-debates', { params }) as Promise<Record<string, unknown>>;
  }

  /**
   * Get a specific matrix debate.
   * @route GET /api/v1/matrix-debates/:id
   */
  async get(debateId: string): Promise<Record<string, unknown>> {
    return this.client.request('GET', `/api/v1/matrix-debates/${debateId}`) as Promise<Record<string, unknown>>;
  }

  /**
   * Create a new matrix debate.
   * @route POST /api/v1/matrix-debates
   */
  async create(data: Record<string, unknown>): Promise<Record<string, unknown>> {
    return this.client.request('POST', '/api/v1/matrix-debates', { json: data }) as Promise<Record<string, unknown>>;
  }
}

File: sdk/python/aragora_sdk/namespaces/matrix_debates.py
"""
Matrix Debates Namespace API

Provides methods for matrix debate management:
- Create and list matrix debates
- Get individual matrix debate details
"""

from __future__ import annotations

from typing import TYPE_CHECKING, Any

if TYPE_CHECKING:
    from ..client import AragoraAsyncClient, AragoraClient


class MatrixDebatesAPI:
    """Synchronous Matrix Debates API."""

    def __init__(self, client: AragoraClient):
        self._client = client

    def list(self, **kwargs: Any) -> dict[str, Any]:
        """
        List matrix debates.

        GET /api/v1/matrix-debates

        Returns:
            Dict with matrix debates list
        """
        return self._client.request("GET", "/api/v1/matrix-debates", params=kwargs or None)

    def get(self, debate_id: str) -> dict[str, Any]:
        """
        Get a matrix debate by ID.

        GET /api/v1/matrix-debates/:id

        Args:
            debate_id: Matrix debate identifier

        Returns:
            Dict with matrix debate details
        """
        return self._client.request("GET", f"/api/v1/matrix-debates/{debate_id}")

    def create(self, data: dict[str, Any]) -> dict[str, Any]:
        """
        Create a new matrix debate.

        POST /api/v1/matrix-debates

        Args:
            data: Matrix debate configuration

        Returns:
            Dict with created matrix debate
        """
        return self._client.request("POST", "/api/v1/matrix-debates", json=data)


class AsyncMatrixDebatesAPI:
    """Asynchronous Matrix Debates API."""

    def __init__(self, client: AragoraAsyncClient):
        self._client = client

    async def list(self, **kwargs: Any) -> dict[str, Any]:
        """List matrix debates. GET /api/v1/matrix-debates"""
        return await self._client.request("GET", "/api/v1/matrix-debates", params=kwargs or None)

    async def get(self, debate_id: str) -> dict[str, Any]:
        """Get a matrix debate by ID. GET /api/v1/matrix-debates/:id"""
        return await self._client.request("GET", f"/api/v1/matrix-debates/{debate_id}")

    async def create(self, data: dict[str, Any]) -> dict[str, Any]:
        """Create a new matrix debate. POST /api/v1/matrix-debates"""
        return await self._client.request("POST", "/api/v1/matrix-debates", json=data)
